### DISCUSSION TITLE
How to serve static generated pages via CDN with Next.js (9.5.2)

### DISCUSSION THREAD

**Discussion by xian0831** [vercel/next.js#16278] [Help]
I'm trying to use Next.js (9.5.2) for both Server Side Rendering and Static Site Generation. SSR works fine for me with assetPrefix. I am able to host all my static assets on CloudFront. However, I'm not sure what the best way is to host the static pages. 

I am facing 2 issues.

 1. `assetPrefix` is not applied to SSG pages. so the link to JS/CSS will be something like this `<link rel="preload" href="/_next/static/css/styles.31b6de8d.chunk.css" as="style"/>`
 2. Assume we host generated HTML on CDN and we are able to serve assets with `assetPrefix`, how do I use Next.js Incremental Static Regeneration with `fallback: true` in `getStaticPath`. My understanding is that page will generated on the server side if the corresponding HTML is not found. 

Thanks everyone for helping.

FYI. I also post question on SO. https://stackoverflow.com/questions/63459231/how-to-serve-static-generated-pages-via-cdn-with-next-js

**Comment by benbachhuber** [comment]
@xian0831 Did you find a solution? This is something I was wondering about as well

**Reply by xian0831** [reply]
@BenBach unfortunately, I didn't find a solution. We end up using SSR for our pages for now. Those pages don't have much traffic, so we haven't run into any issue with SSR. However, we do have a layer of in-memory cache for response coming back from Contenful.

**Reply by xian0831** [reply]
@BenBach I made some progress in my project. You can take a look at what I have tried. I hope it helps 😄 

https://stackoverflow.com/questions/63459231/how-to-serve-static-generated-pages-via-cdn-with-next-js/64995033#64995033

**Reply by kumard3** [reply]
Did you find it now  ?

**Reply by codlin** [reply]
Can the official give a solution?

**Reply by xian0831** [reply]
I found the solution for the my first issue but I'm not sure it is the best solution. We basically just run the build process twice for stage and prod environments.

**Comment by StitiFatah** [comment]
Any followup on that ? Is this possible to serve pages generated with ISR from a CDN if we aren't using Vercel ? The docs only say to upload .next/static to the CDN and not the rest which includes .next/server.

 If no what's the reason ?

**Reply by xian0831** [reply]
It _might be possible_, but I suspect it requires a complex solution to make it work. In theory, you can have a process that watch files that being generated and sync up with the CDN. The invalidation time for those files need to be in sync with what you have in the next.js app. 

I honestly think ISR is basically act as a cache for SSR pages and it is doing a fine job for what it's designed for.

I can't speak for the Vercel team on the reason, but I personally believe that the solution will be super complex and not stable. There are many CDN providers and it is intensive to make sure it works with all of them all the time. Just my 2 cents.
